components/datepicker/DatePickerInput.js

```diff
@@ -0,0 +1,173 @@
+    const { className, dayPickerProps, disabled, modifiers, readOnly, size, ...others } = this.props;
+    const { selectedDate } = this.state;
+
+    const dayPickerClassNames = cx(theme['date-picker'], theme[`is-${size}`], className);
+
+    const rest = omit(others, ['helpText', 'meta', 'onBlur', 'onChange', 'onFocus']);
+
+    return (
+      <DayPickerInput
+        classNames={theme}
+        dayPickerProps={{
+          className: dayPickerClassNames,
+          classNames: theme,
+          modifiers: convertModifiersToClassnames(modifiers, theme),
+          onDayClick: this.handleInputDateChange,
+          selectedDays: selectedDate,
+          navbarElement: <NavigationBar size={size} />,
+          weekdayElement: <WeekDay size={size} />,
+          ...dayPickerProps,
+        }}
+        onDayChange={this.handleInputDateChange}
+        hideOnDayClick={false}
+        inputProps={{
+          disabled: disabled || readOnly,
+          onBlur: this.handleBlur,
+          onFocus: this.handleFocus,
+        }}
+        {...rest}
+      />
+    );
+  };
+
+  renderIcon = () => {
+    const { inverse } = this.props;
+
+    return (
+      <Icon
+        className={theme['input-icon']}
+        color={inverse ? 'teal' : 'neutral'}
+        tint={inverse ? 'light' : 'darkest'}
+        marginHorizontal={2}
+      >
+        <IconCalendarSmallOutline />
+      </Icon>
+    );
+  };
+
+  renderHelpText = () => {
+    const { helpText, inverse } = this.props;
+
+    if (helpText) {
+      return (
+        <TextSmall color={inverse ? 'white' : 'neutral'} marginTop={1} soft>
+          {helpText}
+        </TextSmall>
+      );
+    }
+  };
+
+  renderValidationMessage = () => {
+    return (
+      <Box marginTop={2}>
+        <IconWarningBadgedSmallFilled className={theme['validation-icon']} />
+        <TextSmall className={theme['validation-text']} element="span" marginLeft={1}>
+          {this.props.meta.error}
+        </TextSmall>
+      </Box>
+    );
+  };
+
+  render() {
+    const { bold, disabled, inverse, readOnly, size } = this.props;
+    const { inputHasFocus } = this.state;
+
+    const classNames = cx(theme['date-picker-input'], theme[`is-${size}`], {
+      [theme['is-bold']]: bold,
+      [theme['is-disabled']]: disabled,
+      [theme['is-inverse']]: inverse,
+      [theme['is-read-only']]: readOnly,
+      [theme['has-error']]: this.hasError(),
+      [theme['has-focus']]: inputHasFocus,
+    });
+
+    return (
+      <Box className={classNames}>
+        <div className={theme['input-wrapper']}>
+          {this.renderIcon()}
+          {this.renderDayPickerInput()}
+        </div>
+        {this.hasError() ? this.renderValidationMessage() : this.renderHelpText()}
+      </Box>
+    );
+  }
+}
+
+DatePickerInput.propTypes = {
+  bold: PropTypes.bool,
+  className: PropTypes.string,
+  dayPickerProps: PropTypes.object,
+  disabled: PropTypes.bool,
+  helpText: PropTypes.string,
+  inverse: PropTypes.bool,
```

Comment:
PP: Maybe rename this to isInverted?

Reply:
or isInverse like the css classname

Reply:
Currently we don't use prefixes `is` or `has` for prop names. For now it's consistent to the other components.

 :ok: by @driesd on 06/06/2018 10:39:17

Reply:
We use `inverse` throughout the whole UI, so I would leave it as it is.
we also don't use the `is`-naming for these kind props. Otherwise it would also need to be `isDisabled`, `isReadOnly`, ... I can't really explain why, but it feels better to me to have them without the `is`.

Reply:
Because we're used of that from HTML 👍 

 :ok: by @nickwaelkens on 06/06/2018 11:31:49